### PR TITLE
392: Cache Graphite metrics

### PR DESCRIPTION
**Pull request by PiotrGoralczyk**
This implements caching for #392 and is based upon changes in #381

### CODE REVIEW

**Comment by adamdubiel** [comment]
Please rebase this branch so it contains only Cache code.

**Comment by PiotrGoralczyk** [comment]
@adamdubiel Done.

File: hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/infrastructure/graphite/CachingGraphiteClientTest.groovy [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        cachingClient.readMetrics("metric_1", "metric_2")
+
+        then:
+        1 * underlyingClient.readMetrics("metric_1", "metric_2") >> new GraphiteMetrics([metric_1: "1", metric_2: "2"])
+    }
+
+    def "should get metrics from the underlying client after TTL expires"() {
+        when:
+        cachingClient.readMetrics("metric_1", "metric_2")
+        ticker.advance(Duration.ofSeconds(31))
```
Comment (pbetkier) [inline hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/infrastructure/graphite/CachingGraphiteClientTest.groovy]: nit-picking: you could refer to `CACHE_TTL_IN_SECONDS` here, like `Duration.ofSeconds(CACHE_TTL_IN_SECONDS + 1)`.

Or replace `CACHE_TTL_IN_SECONDS` constant with sth like `CACHE_TTL` of type `Duration`, making this call like this: `CACHE_TTL.plusSeconds(1)`.

Similarly in the test before.

**Comment by pbetkier** [comment]
@PiotrGoralczyk nice & simple.

:+1:

**Comment by adamdubiel** [comment]
:+1:

**Comment by PiotrGoralczyk** [comment]
Implemented @pbetkier 's comment, ready to merge.

**Pull request merged**